### PR TITLE
Move to v2 endpoint

### PR DESCRIPTION
**Pull request by joseartrivera**
This PR swaps out the Store API endpoint from the v1 version to the v2 version.

- Changes the endpoint from https://manage.devcenter.microsoft.com to https://api.partner.microsoft.com for both the StoreBroker client and RestProxy service.
- Updated the proxy to pass down the new Request-ID header and updated the client to log the Request-ID header.
- Removed the MS-CorrelationId header, the API now uses the MS-ClientRequestId header to track multiple API calls. The logic for automatically generating a CorrelationId if one was not passed was re-purposed to generate a ClientRequestId instead.
- Updates the RestProxy Security Protocol to be TLS 1.2 as that is what the new endpoint expects.

### CODE REVIEW

File: StoreBroker/StoreIngestionApi.psm1 [hunk 1]
```diff
@@ -51,12 +51,11 @@ $script:headerClientName = 'X-ClientName'
 
 # Special header added to Submission API responses that provides a unique ID
 # that the Submission API team can use to trace back problems with a specific request.
-$script:headerMSRequestId = 'MS-RequestId'
+$script:headerRequestId = 'Request-ID'
```
Comment (HowardWolosky) [inline StoreBroker/StoreIngestionApi.psm1:54]: Same request to confirm capitalization

File: StoreBroker/StoreIngestionApi.psm1 [hunk 2]
```diff
@@ -51,12 +51,11 @@ $script:headerClientName = 'X-ClientName'
 
 # Special header added to Submission API responses that provides a unique ID
 # that the Submission API team can use to trace back problems with a specific request.
-$script:headerMSRequestId = 'MS-RequestId'
+$script:headerRequestId = 'Request-ID'
 
 # Special headers that clients can add to their requests to make it easier to track the
```
Comment (HowardWolosky) [inline StoreBroker/StoreIngestionApi.psm1:56]:
```suggestion
# Special header that clients can add to their requests to make it easier to track the
```

File: StoreBroker/StoreIngestionApi.psm1 [hunk 3]
```diff
@@ -1894,17 +1888,11 @@ function Invoke-SBRestMethod
             }
         }
 
-        # Add any headers that clients can add to track specific requests
```
Comment (HowardWolosky) [inline StoreBroker/StoreIngestionApi.psm1:1897]: Why remove this comment?
```suggestion
        # Add a header that clients can add to track specific requests
```

File: RESTProxy/Models/Endpoint.cs [hunk 4]
```diff
@@ -366,15 +357,17 @@ private HttpResponseMessage GetResponseMessage(HttpWebResponse httpResponseFromA
             }
 
             // Proxy all of the special headers that the API returns
-            // (which all begin with "MS-").  One example is "MS-CorrelationId"
+            // (which all begin with "MS-").  One example is "MS-Client-RequestId"
             // which is needed by the Windows Store Submission API team when they
             // are investigating bug reports with the API.
             const string MSHeaderPreface = "MS-";
+            const string RequestHeaderPreface = "Request-";
```
[the remark this answers is not in the document]
Comment (joseartrivera) [reply RESTProxy/Models/Endpoint.cs]: Updated

File: StoreBroker/ConvertFrom-ExistingSubmission.ps1 [hunk 5]
```diff
@@ -242,8 +239,7 @@ function ConvertFrom-ExistingSubmission
             [StoreBrokerTelemetryProperty]::ProductId = $ProductId
             [StoreBrokerTelemetryProperty]::AppId = $AppId
             [StoreBrokerTelemetryProperty]::SubmissionId = $submissionId
-            [StoreBrokerTelemetryProperty]::ClientRequestId = $ClientRequesId
```
[the remark this answers is not in the document]
Comment (joseartrivera) [reply StoreBroker/ConvertFrom-ExistingSubmission.ps1:245]: I actually didn't notice the typo either, I just deleted the old ClientRequestID and replaced the existing correlationID with ClientRequestID :P

File: StoreBroker/StoreIngestionSubmissionApi.ps1 [hunk 6]
```diff
@@ -341,10 +333,6 @@ function New-Submission
         until all of its resources are "ready".
 
     .PARAMETER ClientRequestId
-        An optional identifier that should be sent along to the Store to help with identifying
-        this request during post-mortem debugging.
-
-    .PARAMETER CorrelationId
         An optional identifier that should be sent along to the Store to help with identifying
         this request during post-mortem debugging.  This is typically supplied when trying to
```
[the remark this answers is not in the document]
Comment (joseartrivera) [reply StoreBroker/StoreIngestionSubmissionApi.ps1:337]: Yes, ClientRequestId is the new CorrelationId

File: RESTProxy/Models/ProxyManager.cs [hunk 7]
```diff
@@ -29,16 +29,10 @@ public static class ProxyManager
...
-        ///  The header name for the CorrelationId that the user or API sets for tracking a series of related requests
-        ///  for post-mortem diagnostics.
-        /// </summary>
-        public const string MSCorrelationIdHeader = "MS-CorrelationId";
-
         /// <summary>
         ///  The header name for the RequestId that the API adds for post-mortem diagnostics.
         /// </summary>
-        public const string MSRequestIdHeader = "MS-RequestId";
+        public const string RequestIdHeader = "Request-ID";
```
[the remark this answers is not in the document]
Comment (joseartrivera) [reply RESTProxy/Models/ProxyManager.cs:35]: This is the capitalization used when returned by the API. Also headers are case insensitive.

File: StoreBroker/StoreIngestionApi.psm1 (hunk 3 above)
Comment (joseartrivera) [reply StoreBroker/StoreIngestionApi.psm1:1897]: Added back

File: RESTProxy/Models/ProxyManager.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (HowardWolosky) [reply RESTProxy/Models/ProxyManager.cs:35]: I know that their case-insensitive, but it helps to be consistent.  Good to know that this is the casing that they're using.

**Pull request merged**